modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,96 @@
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 0);
+    });
+
+    it('TINY-4612: caret wont move if it is not at inline element at the start of the line', async function () {
+      // Page Up/Down on Mac scroll the viewport instead of change the caret position/selection
+      if (!platform.os.isOSX()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link2</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 4);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 4);
+    });
+
+    it('TINY-4612: caret wont move inline_boundaries: false', async function () {
+      if (!platform.os.isOSX()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 1 ], 2);
+      delete editor.settings.inline_boundaries;
+    });
+  });
+
+  context('Page Down', () => {
+    it('TINY-4612: caret should be placed at the end of the line if the last child is an inline element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: caret wont move if it is not at inline element at the end of the line', async function () {
+      // Windows os default behavior (regardless the browser) is moving the caret at the end of the line
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 1 ], 2);
+    });
+
+    it('TINY-4612: caret wont move inline_boundaries: false', async function () {
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
```

Comment:
Another assertion that's not what should be happening, similar to the above.

Reply:
updated